agent/src/main/java/com/appland/appmap/util/AppMapClassPool.java

```diff
@@ -0,0 +1,37 @@
+package com.appland.appmap.util;
+
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.function.Supplier;
+
+import javassist.ClassPool;
+import javassist.LoaderClassPath;
+
+public class AppMapClassPool {
+  private static Supplier<ClassPool> builder;
+  private static final ThreadLocal<Deque<ClassPool>> threadPoolQueue =
+      ThreadLocal.withInitial(() -> new ArrayDeque<ClassPool>());
+
+  // public static ClassPool acquire() {
+  // return acquire(Thread.currentThread().getContextClassLoader());
+  // }
+
+  public static ClassPool acquire(ClassLoader classLoader) {
+    ClassPool ret = builder != null ? builder.get() : new ClassPool();
+    ret.appendClassPath(new LoaderClassPath(classLoader));
+    threadPoolQueue.get().push(ret);
+    return ret;
+  }
+
+  public static ClassPool get() {
+    return threadPoolQueue.get().peek();
```

Comment:
Would it make sense to assert it's not empty (also in `release()`), or is it enough to throw NPE down the line?

Reply:
I added the check in `get`. In `release`, `pop` will throw an exception if
the Deque is empty, which seems sufficient.